flink-architecture-tests/src/test/java/org/apache/flink/architecture/rules/ITCaseRules.java

```diff
@@ -0,0 +1,91 @@
+
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static com.tngtech.archunit.core.domain.JavaModifier.ABSTRACT;
+import static com.tngtech.archunit.library.freeze.FreezingArchRule.freeze;
+import static org.apache.flink.architecture.common.Conditions.fulfill;
+import static org.apache.flink.architecture.common.GivenJavaClasses.javaClassesThat;
+import static org.apache.flink.architecture.common.Predicates.arePublicFinalOfTypeWithAnnotation;
+import static org.apache.flink.architecture.common.Predicates.arePublicStaticFinalOfTyp;
+import static org.apache.flink.architecture.common.Predicates.arePublicStaticFinalOfTypeWithAnnotation;
+import static org.apache.flink.architecture.common.Predicates.containAnyFieldsInClassHierarchyThat;
+
+/** Rules for Integration Tests. */
+public class ITCaseRules {
+
+    @ArchTest
+    public static final ArchRule INTEGRATION_TEST_ENDING_WITH_ITCASE =
+            freeze(
+                    javaClassesThat()
+                            .areAssignableTo(AbstractTestBase.class)
+                            .and()
+                            .doNotHaveModifier(ABSTRACT)
+                            .should()
+                            .haveSimpleNameEndingWith("ITCase"));
+
+    @ArchTest
+    public static final ArchRule ITCASE_USE_MINICLUSTER =
+            freeze(
+                    javaClassesThat()
+                            .haveSimpleNameEndingWith("ITCase")
+                            .and()
+                            .areTopLevelClasses()
+                            .and()
+                            .doNotHaveModifier(ABSTRACT)
+                            .should(
+                                    fulfill(
+                                            // JUnit 5 violation check
+                                            containAnyFieldsInClassHierarchyThat(
+                                                            arePublicStaticFinalOfTyp(
+                                                                    MiniClusterExtension.class))
+                                                    .and(
+                                                            containAnyFieldsInClassHierarchyThat(
+                                                                    arePublicStaticFinalOfTypeWithAnnotation(
+                                                                            AllCallbackWrapper
+                                                                                    .class,
+                                                                            RegisterExtension
+                                                                                    .class)))
+                                                    // JUnit 4 violation check, which should be
```

Comment:
Good idea, I will create a task under the JUnit migration umbrella. I would recommend to have this comment in the code for every JUnit 4 rule. Otherwise it will be a big effort to clean it up in the future.